docs/weave/tutorial/11-cron.md

```diff
@@ -0,0 +1,325 @@
+inaccesible to the world via a query so we can implement our execution without any fears of
+exploitation.
+
+This handler will receive the `DeleteArticleMsg`, delete the article from the store using the
+article ID data that is passed along with the message.
+
+## Task Cancellation
+
+If we need the ability to do execution in distant future, we also need the ability to cancel the
+same execution. As you remember we saved the scheduled task information to the store to be used
+in various post-submission activities. Deletion is the simplest example of this case. As you read
+many many times, and probably got very used to, every action in Weave goes through an handler. We
+will create a handler that is called `CancelDeleteArticleTaskHandler`.
+
+`validate` method:
+```go
+// validate does all common pre-processing between Check and Deliver
+func (h CancelDeleteArticleTaskHandler) validate(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*CancelDeleteArticleTaskMsg, error) {
+	var msg CancelDeleteArticleTaskMsg
+
+	if err := weave.LoadMsg(tx, &msg); err != nil {
+		return nil, errors.Wrap(err, "load msg")
+	}
+
+	var task DeleteArticleTask
+	if err := h.b.One(store, msg.TaskID, &task); err != nil {
+		return nil, errors.Wrapf(err, "delete task with id %s not found", msg.TaskID)
+	}
+
+	signer := x.MainSigner(ctx, h.auth).Address()
+	if !task.TaskOwner.Equals(signer) {
+		return nil, errors.Wrapf(errors.ErrUnauthorized, "signer %s is unauthorized to cancel scheduled delete article task with id %s", signer, msg.TaskID)
+	}
+
+	return &msg, nil
+}
+```
+
+Since cancellation of tasks requries to be initiated by a user signed message, we need to
+explicitly check if user signature is in the context, same as every other regular handler.
+
+If the task desired to be cancelled exists and task owner is the signer of the address, execution
+proceeds to the `Deliver` method:
+
+```go
+// Deliver cancels delete task if conditions are met
+func (h CancelDeleteArticleTaskHandler) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
+	msg, err := h.validate(ctx, store, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := h.scheduler.Delete(store, msg.TaskID); err != nil {
+		return nil, errors.Wrapf(err, "cannot delete scheduled task with id %s", msg.TaskID)
+	}
+
+	if err := h.b.Delete(store, msg.TaskID); err != nil {
+		return nil, errors.Wrapf(err, "cannot cancel delete task with id %s", msg.TaskID)
+	}
+
+	return &weave.DeliverResult{Data: msg.TaskID}, nil
+}
+```
+
+First, task gets **descheduled** using the task ID present in the message and then deleted from
+the task store.
+
+## CRON Stack
+
+As we mentioned before, CRON handlers are no different than regular handlers but they differ in
+one point: CRON operates in a totally different stack than the application.
+
+We need to define a seperate routes for our CRON stack in the blog module:
```

Comment:
Instead of _we need_ I would write _we can_ as this is not mandatory, but a good practice.